tests/test_scan.py

```diff
@@ -0,0 +1,180 @@
+import asyncio
+
+import pytest
+
+from aioelasticsearch import NotFoundError
+from aioelasticsearch.helpers import Scan
+from tests.utils import populate
+
+
+def test_scan_initial_raises(loop, es):
+    scan = Scan(es, loop=loop)
+
+    with pytest.raises(AssertionError):
+        scan.scroll_id
+
+    with pytest.raises(AssertionError):
+        scan.total
+
+    with pytest.raises(AssertionError):
+        scan.has_more
+
+    with pytest.raises(AssertionError):
+        next(scan)
+
+    with pytest.raises(AssertionError):
+        loop.run_until_complete(scan.search())
+
+
+@pytest.mark.parametrize('n,scroll_size', [
+    (0, 1),  # no results
+    (6, 6),  # 1 scroll
+    (6, 8),  # 1 scroll
+    (6, 3),  # 2 scrolls
+    (6, 4),  # 2 scrolls
+    (6, 2),  # 3 scrolls
+    (6, 1),  # 6 scrolls
+])
+@pytest.mark.run_loop
+@asyncio.coroutine
+def test_scan_equal_chunks_for_loop(loop, es, n, scroll_size):
+    index = 'test_aioes'
+    doc_type = 'type_1'
+    yield from populate(es, index, doc_type, n, loop=loop)
+
+    ids = set()
+    data = []
+    with Scan(
+        es,
+        index=index,
+        doc_type=doc_type,
+        size=scroll_size,
+        loop=loop,
+    ) as scan:
+        yield from scan.scroll()
+
+        for scroll in scan:
+            docs = yield from scroll
+
+            if docs:
```

Comment:
if there are no docs in es:
https://github.com/litwisha/aioelasticsearch/blob/scan-tests-and-fixes/tests/test_scan.py#L30